cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/DefaultPluginContext.java

```diff
@@ -99,8 +99,19 @@ public PluginProperties getPluginProperties(String pluginId) {
 
   @Override
   public <T> T newPluginInstance(String pluginId, MacroEvaluator evaluator) throws InstantiationException {
-    // todo : use pluginInstantiator#newPluginInstance(String pluginId, MacroEvaluator evaluator)
-    return newPluginInstance(pluginId);
+    try {
+      Plugin plugin = getPlugin(pluginId);
+      if (pluginInstantiator == null) {
+        throw new UnsupportedOperationException("Plugin is not supported");
+      }
+      return pluginInstantiator.newInstance(plugin, evaluator);
+    } catch (ClassNotFoundException e) {
+      // Shouldn't happen, unless there is bug in file localization
+      throw new IllegalArgumentException("Plugin class not found", e);
```

Comment:
InstantiationException won't take a throwable in its constructor, is it ok to lose the ClassNotFoundException?